src/forms/TextInput.jsx

```diff
@@ -69,6 +71,17 @@ export const TextInput = (props) => {
 	const inputStyles = iconShape && {
 		paddingLeft: `${paddingSize}px`
 	};
+	const customValidityMessage = isValid ? '' : validityMessage;
+
+	let textInput = null;
+
+	const handleOnChange = (e) => {
+		if (onChange) {
+			onChange(e);
+		}
+
+		textInput && textInput.setCustomValidity(customValidityMessage);
```

Comment:
why are we checking for `textInput` before setting it's validity message?

Reply:
Tests were failing because of setting the textInput to null